mmv1/third_party/terraform/tests/resource_compute_network_endpoints_test.go.erb

```diff
@@ -0,0 +1,293 @@
+<% autogen_exception -%>
+package google
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func TestAccComputeNetworkEndpoints_networkEndpointsBasic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": RandString(t, 10),
+		"default_port":  90,
+		"modified_port": 100,
+		"add1_port":     101,
+		"add2_port":     102,
+		"add3_port":     103,
+		"add4_port":     104,
+        "add5_port":     105,
+	}
+	negId := fmt.Sprintf("projects/%s/zones/%s/networkEndpointGroups/tf-test-neg-%s",
+		acctest.GetTestProjectFromEnv(), acctest.GetTestZoneFromEnv(), context["random_suffix"])
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				// Create one endpoint
+				Config: testAccComputeNetworkEndpoints_networkEndpointsBase(context),
+			},
+			{
+				ResourceName:      "google_compute_network_endpoints.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Force-recreate old endpoint
+				Config: testAccComputeNetworkEndpoints_networkEndpointsModified(context),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeNetworkEndpointsWithPortsDestroyed(t, negId, "90"),
+				),
+			},
+			{
+				ResourceName:      "google_compute_network_endpoints.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Add four new endpoints
+				Config: testAccComputeNetworkEndpoints_networkEndpointsAdditional(context),
+			},
+			{
+				ResourceName:      "google_compute_network_endpoints.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Add enough endpoints to trigger pagination
+				Config: testAccComputeNetworkEndpoints_networkEndpointsPaginated(context, 100, 1300),
+			},
+			{
+				ResourceName:      "google_compute_network_endpoints.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Remove enough endpoints to trigger pagination
+				Config: testAccComputeNetworkEndpoints_networkEndpointsPaginated(context, 700, 1900),
+			},
+			{
+				ResourceName:      "google_compute_network_endpoints.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// delete all endpoints
+				Config: testAccComputeNetworkEndpoints_noNetworkEndpoints(context),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeNetworkEndpointsWithPortsDestroyed(t, negId, "100"),
```

Comment:
Since we're dealing with pagination and an authoritative resource, this should check that there are zero network endpoints left (rather than checking a specific one was destroyed.)

Reply:
Ack, added a new function to check there aren't any endpoints left.